src/Unused/Util.hs

```diff
@@ -1,12 +1,22 @@
 module Unused.Util
     ( groupBy
+    , stringToInt
     ) where
 
-import Data.List (nub)
+import Control.Arrow ((&&&))
+import qualified Data.List as L
+import Data.Function
+import Data.Char (digitToInt, isDigit)
 
-groupBy :: Eq b => (a -> b) -> [a] -> [(b, [a])]
-groupBy f l =
-    fmap (\t -> (t, byTerm t)) uniqueTerms
+groupBy :: (Ord b) => (a -> b) -> [a] -> [(b, [a])]
+groupBy f = map (f . head &&& id)
+                   . L.groupBy ((==) `on` f)
+                   . L.sortBy (compare `on` f)
+
+stringToInt :: String -> Maybe Int
+stringToInt xs
+    | all isDigit xs = Just $ loop 0 xs
+    | otherwise = Nothing
   where
-    byTerm t = filter ((== t) . f) l
-    uniqueTerms = nub $ fmap f l
+    loop acc [] = acc
```

Comment:
@jferris nice catch, you're right - updated in efcb46a285fabeb5ac4f578cc382cf0f0d1fff66